src/dataflow/decode/mod.rs

```diff
@@ -118,7 +118,284 @@ where
+            stream,
+            OffsetDecoderState {
+                datum_func: bytes_to_datum,
+            },
+            avro::AvroDecoderState::new(
+                &val_enc.value_schema,
+                val_enc.schema_registry_url,
+                interchange::avro::EnvelopeType::Upsert,
+            ),
+            &op_name,
+        ),
+        (DataEncoding::Text, DataEncoding::Avro(val_enc)) => decode_key_values_inner(
+            stream,
+            OffsetDecoderState {
+                datum_func: text_to_datum,
+            },
+            avro::AvroDecoderState::new(
+                &val_enc.value_schema,
+                val_enc.schema_registry_url,
+                interchange::avro::EnvelopeType::Upsert,
+            ),
+            &op_name,
+        ),
+        (DataEncoding::Avro(key_enc), DataEncoding::Avro(val_enc)) => decode_key_values_inner(
+            stream,
+            avro::AvroDecoderState::new(
+                &key_enc.value_schema,
+                key_enc.schema_registry_url,
+                interchange::avro::EnvelopeType::None,
+            ),
+            avro::AvroDecoderState::new(
+                &val_enc.value_schema,
+                val_enc.schema_registry_url,
+                interchange::avro::EnvelopeType::Upsert,
+            ),
+            &op_name,
+        ),
+        (DataEncoding::Text, DataEncoding::Bytes) => decode_key_values_inner(
+            stream,
+            OffsetDecoderState {
+                datum_func: text_to_datum,
+            },
+            OffsetDecoderState {
+                datum_func: bytes_to_datum,
+            },
+            &op_name,
+        ),
+        (DataEncoding::Bytes, DataEncoding::Bytes) => decode_key_values_inner(
+            stream,
+            OffsetDecoderState {
+                datum_func: bytes_to_datum,
+            },
+            OffsetDecoderState {
+                datum_func: bytes_to_datum,
+            },
+            &op_name,
+        ),
+        (DataEncoding::Text, DataEncoding::Text) => decode_key_values_inner(
+            stream,
+            OffsetDecoderState {
+                datum_func: text_to_datum,
+            },
+            OffsetDecoderState {
+                datum_func: text_to_datum,
+            },
+            &op_name,
+        ),
+        _ => unreachable!(),
+    };
+    decoded_stream.map(|(key, value, timestamp)| {
+        if let Some(value) = value {
+            let mut value_with_key = RowPacker::new();
+            value_with_key.extend_by_row(&key);
+            value_with_key.extend_by_row(&value);
+            (key, Some(value_with_key.finish()), timestamp)
+        } else {
+            (key, None, timestamp)
+        }
+    })
+}
+
+pub fn drop_key<G>(
+    stream: &Stream<G, ((Vec<u8>, Vec<u8>), Option<i64>)>,
+) -> Stream<G, (Vec<u8>, Option<i64>)>
+where
+    G: Scope<Timestamp = Timestamp>,
+{
+    stream.map(|((_key, payload), aux_num)| (payload, aux_num))
+}
```

Comment:
Is this function live? I couldn't find it used in the PR. If it is, it might be better placed in-line (at least, it could be fused with the caller, to avoid the dataflow complexity). Ignore if it is used somewhere and I'm clueless.

Reply:
Ah, I found it sorry (@umanwizard pointed me at the right file to expand, which I hadn't). It's still a bit heavy for a function, but maybe a style thing (closures are meant to make in-line timely operators idiomatic to use).

Reply:
I fused it with the caller.